src/services/utils/get-sorted-dimension-info.ts

```diff
@@ -0,0 +1,26 @@
+import { PSEUDO_DIMENSION_INDEX } from "../../constants";
+import type { ExtendedDimensionInfo, PivotLayout, PseudoDimension } from "../../types/QIX";
+
+const filterPseudoAndVisibleDimensions = (qDim: ExtendedDimensionInfo | -1) =>
+  qDim === PSEUDO_DIMENSION_INDEX || qDim.qCardinalities.qHypercubeCardinal > 0;
+
+const getSortedDimensionInfo = (layout: PivotLayout) => {
+  const { qHyperCube } = layout;
+  const { qNoOfLeftDims, qEffectiveInterColumnSortOrder, qDimensionInfo } = qHyperCube;
+
+  const sortedDimensionInfo = qEffectiveInterColumnSortOrder.map((index) => {
+    if (index === PSEUDO_DIMENSION_INDEX) return PSEUDO_DIMENSION_INDEX;
```

Comment:
I think it would be better to take a solution like defining dim type in `ExtendedDimensionInfo` and then the output of this function will be only `ExtendedDimensionInfo[]` which covers a type of dim called Pseudo. 

That way we could do what ever we want and won't need to check for type of this object where we use it (example in create-cell module in `isLockedByDimension` key)

+ also we can always safely perform higher order functions on it's return value (`sortedDimensionInfo`) without worrying about the type of items in array.

Reply:
Ideally I want the array to be of type `ExtendedDimensionInfo[]` but in some part of the code, there is a need to iterate the sortedDimensionInfo array and be able to tell if an item in the array is a pseudo dimension or not.

What I think I can do is create a new type that is not a integer but an object for the `PseudoDimension` type instead. That way all items in the array would be objects. But I cannot really make it into a `ExtendedDimensionInfo` type since that has lots of required properties that I don't want to fake for the sake of the solution.

Reply:
The thing that may cause problem in that case is even if you have it in object, again when doing a map, filter, ... on the end result, you are going to need to cast iteration item to either that object or `ExtendedDimensionInfo` type and you need to take care of it (but of course it directly ties into how often we want to use that data)

Reply:
I'm not seeing any other alternative atm. There is a need to have an array that contains both dimensions and a pseudo dimensions. As both type of dimensions are rendered for the user as a "dimension". I would have to separate that somehow.

I can have an array with only numbers, including `-1`, that points to a qDimensionInfo (essentially the `qEffectiveInterColumnSortOrder`). But when iterating that using `map` or `filter` you end up having to check if the number is `-1` anyway and have some special logic for that. So in that sense the end result is the same or perhaps even more complex.